src/database-assignment-schema.sql

```diff
@@ -0,0 +1,183 @@
+(
+    provider   VARCHAR(50),
+    encoding   VARCHAR(50),
+    product_id INT PRIMARY KEY,
+    FOREIGN KEY (product_id) REFERENCES products (id) ON DELETE CASCADE
+);
+
+CREATE TABLE tv_compatible_ci_modules
+(
+    television_id INT,
+    ci_module_id  INT,
+    PRIMARY KEY (television_id, ci_module_id),
+    FOREIGN KEY (television_id) REFERENCES televisions (product_id) ON DELETE CASCADE,
+    FOREIGN KEY (ci_module_id) REFERENCES ci_modules (product_id) ON DELETE CASCADE
+);
+
+CREATE TABLE wall_brackets
+(
+    adjustable      BOOLEAN,
+    mounting_method VARCHAR(50),
+    height          DOUBLE PRECISION NOT NULL,
+    width           DOUBLE PRECISION NOT NULL,
+    product_id      INT PRIMARY KEY,
+    FOREIGN KEY (product_id) REFERENCES products (id) ON DELETE CASCADE
+);
+
+CREATE TABLE tv_compatible_wall_brackets
+(
+    television_id   INT,
+    wall_bracket_id INT,
+    PRIMARY KEY (television_id, wall_bracket_id),
+    FOREIGN KEY (television_id) REFERENCES televisions (product_id) ON DELETE CASCADE,
+    FOREIGN KEY (wall_bracket_id) REFERENCES wall_brackets (product_id) ON DELETE CASCADE
+);
+
+INSERT INTO products (name, brand, price, current_stock, sold, date_sold, type)
+VALUES ('UltraVision 4K', 'Samsung', 899.99, 15, 5, '2025-05-01', 'television'),
+       ('SlimOLED 55', 'LG', 1299.00, 5, 2, '2025-05-05', 'television'),
+       ('SmartDisplay 32', 'Philips', 299.99, 20, 0, NULL, 'television'),
+       ('SmartRemote X', 'Samsung', 39.99, 50, 25, '2025-05-10', 'remote_controller'),
+       ('Basic Remote A', 'LG', 19.99, 80, 0, NULL, 'remote_controller'),
+       ('CI+ Module Ziggo', 'Ziggo', 49.99, 30, 12, '2025-04-29', 'ci_module'),
+       ('CI+ Module KPN', 'KPN', 44.95, 25, 0, NULL, 'ci_module'),
+       ('WallFlex Tilt', 'Vogel''s', 79.99, 10, 5, '2025-04-30', 'wall_bracket'),
+       ('SlimMount Fixed', 'Philips', 29.99, 40, 0, NULL, 'wall_bracket');
+
+INSERT INTO remote_controllers (product_id, smart, battery_type)
+VALUES (4, true, 'AAA'),
+       (5, false, 'AA');
+
+INSERT INTO televisions (product_id, height, width, image_quality, screen_type, wifi, smart_tv, voice_control, hdr,
+                         remote_id)
+VALUES (1, 70.0, 110.0, '4K UHD', 'LED', true, true, true, true, 4),
+       (2, 65.0, 100.0, '4K OLED', 'OLED', true, true, true, true, 5),
+       (3, 40.0, 70.0, 'HD', 'LCD', true, false, false, false, NULL);
+
+INSERT INTO ci_modules (product_id, provider, encoding)
+VALUES (6, 'Ziggo', 'MPEG4'),
+       (7, 'KPN', 'H.264');
+
+INSERT INTO wall_brackets (product_id, adjustable, mounting_method, height, width)
+VALUES (8, true, 'tilt', 20.0, 40.0),
+       (9, false, 'fixed', 15.0, 35.0);
+
+INSERT INTO tv_compatible_ci_modules (television_id, ci_module_id)
+VALUES (1, 6),
+       (1, 7),
+       (2, 6);
+
+INSERT INTO tv_compatible_wall_brackets (television_id, wall_bracket_id)
+VALUES (1, 8),
+       (2, 8),
+       (2, 9),
+       (3, 9);
+
+-- SELECTION STATEMENTS
+SELECT * FROM products;
```

Comment:
Deze is helemaal in lijn met onze les en Edhub, maar in de praktijk is het beter om queries specifiek op te schrijven (zoals je hieronder heel mooi doet). Reden hiervoor is volgens mij met name performance en minder foutgevoeligheid.